### PR TITLE
11933 - At-Start stacks were getting confused by blank boards

### PR DESCRIPTION
**Pull request by Cattlesquat**
This might fix the problem Benkyo was experiencing (which manifested while dragging SetupStacks in and out of folders, but really was about blank stupid boards)

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java [hunk 1]
```diff
@@ -2676,6 +2676,7 @@ public boolean importData(TransferHandler.TransferSupport support) {
 
           if (remove(getParent(sourceNode), cutObj)) {
             insert(target, convertedCutObj, childIndex);
+            postPasteFixups(convertedCutObj);
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java:2679]: HA, This is the actual fix. I don't completely missed that in the other code paths!

File: vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java [hunk 2]
```diff
@@ -386,6 +386,9 @@ else if (OWNING_BOARD.equals(key)) {
           owningBoardName = null;
         }
       }
+      else if (value.equals("")) {
+        owningBoardName = null; // Because "" is what null sometimes turns into during Vassal configurers
+      }
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java:391]: I don't believe this is needed once you add the postPasteFixes() call in. Just adding the postPasteFixes() call fixed the issue for me.

**Review by BrentEaston — CHANGES_REQUESTED** [review]

**Comment by BrentEaston** [comment]
I don't see any problem with this going into 3.6.11rather than 3.7.0? It's basically a just a bug fix.

File: vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java (hunk 2 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java:391]: Still needed -- see comment above.

File: vassal-app/src/main/resources/VASSAL/i18n/Editor.properties [hunk 3]
```diff
@@ -301,6 +301,7 @@ Editor.Board.reverse=Reversible
 Editor.Board.width=Board width
 Editor.Board.height=Board height
 Editor.Board.component_type=Board
+Editor.Board.main_board=Main Board
```
[the remark this answers is not in the document]
Comment (Cattlesquat) [reply vassal-app/src/main/resources/VASSAL/i18n/Editor.properties]: Fixed

File: vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java (hunk 1 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java:2679]: In the "drag" version, this was (on the one hand) "fixing" the problem, but it was creating a new "bad data" message in the console -- in other words, it WASN'T the thing making the cut/paste thing work. Cut/Paste already didn't generate the problem and so didn't get the extra warning message. So this DOES belong here (for other reasons) but DOESN'T really fix this problem.

**Comment by Cattlesquat** [comment]
Agreed it could go in 3.6.11 if we do one

**Comment by BrentEaston** [comment]
Looks good to me. I've asked Benkyo to do a final check.

File: vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java [hunk 4]
```diff
@@ -386,6 +386,9 @@ else if (OWNING_BOARD.equals(key)) {
           owningBoardName = null;
         }
       }
+      else if (value.equals("")) {
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java:389]: You should  check `"".equals(value)` instead. This way, you're still safe if `value` is null.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/SetupStack.java]: Fixed

**Pull request merged**